src/prisma/__mocks__/prisma.service.mock.ts

```diff
@@ -48,6 +58,61 @@ export class MockPrismaService {
         email: faker.internet.email(),
         password: faker.internet.password(),
         role: Role.USER,
+        activated: false,
+        accountActivationToken: null,
+      });
+    },
+  };
+
+  pendingUser = {
+    create: function (request: {
+      data: Prisma.UserCreateInput;
+    }): Promise<PendingUser> {
+      return Promise.resolve({
+        id: faker.number.int(),
+        email: request.data.email,
+        password: request.data.password,
+        accountActivationToken: faker.string.alphanumeric(32),
+        createdAt: new Date(),
+      });
+    },
+
+    update: function (request: {
+      where: { id: number };
+      data: {
+        email?: string;
+        password?: string;
+        accountActivationToken?: string;
+      };
+    }): Promise<PendingUser> {
+      const { email, password, accountActivationToken } = request.data;
+      return Promise.resolve({
+        id: request.where.id,
+        email: email ?? faker.internet.email(),
+        password: password ?? faker.internet.password(),
+        createdAt: new Date(),
+        accountActivationToken: accountActivationToken ?? null,
+      });
+    },
+
+    delete: function (_request: { id: number }): Promise<void> {
+      return Promise.resolve();
+    },
+
+    findUnique: function (request: {
+      where: { id?: number; email?: string };
+    }): Promise<PendingUser> | Promise<void> {
+      const { id, email } = request.where;
+
+      if (email) {
+        return Promise.resolve();
+      }
```

Comment:
hmm, I don't quite understand this mock - if `email` is present then we don't return anything?

Reply:
Yea, I use this if to be able to get correct response on `signUp` tests - if this `findUnique({where: {email: ... }})` would return something, then user would get 401